### ISSUE TITLE
Search bar loses keyboard focus on submit

### ISSUE THREAD

**Issue by seanh** [hypothesis/h#3984]
Type a word into the search bar, you can either press space to lozengify your word or not, hit `Enter`: form submits and page reloads, but the keyboard focus is no longer in the search bar.

This also happens if you click `x` to delete a lozenge.

Expected: keyboard focus should remain in the search bar.

Even if you don't type something into the search and submit it, when you first load the search page maybe the empty search bar should have keyboard focus automatically?

**Comment by sean-roberts** [comment]
We need to be careful with this. It could cause a really odd mobile experience. 

My suggestion is that we autofocus only when we have no current lozenges. My expectation there is that the journey is a user navigating, intending to search. If we are loading the page and there are lozenges, it's possible that we are going to a shared link. This is where the weirdness comes in when the user is not intending to search. The gist is, we can't easily assume that the user is coming there to search in all cases.

**Comment by sean-roberts** [comment]
Additionally, we show the suggestion drop down on initial focus. This is another point that would need to be massaged to not be obtrusive.

**Comment by seanh** [comment]
@robertknight This is fixed right?

**Comment by robertknight** [comment]
@seanh - Yes. I think the approach implemented in https://github.com/hypothesis/h/pull/4088 addresses this, at least for devices that have an always-on keyboard.
